### ISSUE TITLE
Incorrect Chimera API signature

### ISSUE THREAD

**Issue by artyyouth** [flier/rust-hyperscan#26]
The line here https://github.com/flier/rust-hyperscan/blob/b42c421fec7b24ace432c613f72deed76f4a7f2d/hyperscan/src/chimera/runtime.rs#L58

shouldn't it just be `pub fn realloc_scratch(&self, &mut s: Scratch) -> Result<Scratch> {` because we are not mutating the `DatabaseRef`?

**Comment by flier** [comment]
Agree, the API should be 
```rust
 pub fn realloc_scratch(&self, s: &mut Scratch) -> Result<&mut ScratchRef>
```

**Comment by flier** [comment]
fixed in v0.3.1, Thanks
